fip-0026.md

```diff
@@ -339,251 +409,281 @@ This sets the marketplace configurations.
-|Invalid Warn Days|`warn_days` must be a positive number less than the value of `duration`|400| | |"`warn_days` must be a positive number less than the value of `duration`"|
+|Invalid marketplace name|Supplied `marketplace` name is not between 1 and 25 characters|400|"marketplace"|empty string, "", "areallylongmarketplacename" |"Length of marketplace name should be between 1 and 25 characters"|
+|Invalid actor name|Supplied `owner` is not a registered actor on FIO.|400|"owner"| |"Account not registered on FIO"|
+|Invalid public key|Invalid FIO Public Key|400|"owner_public_key"|Value sent in, e.g."FIOinValidKey1234232"|"Invalid Public Key"|
+|Invalid commission fee|`commissionfee` is not between 0-25|400|"comissionfee"|Value sent in, e.g. "50" |"Commission fee can only be between 0 and 25"|
+|Invalid listing fee|`listingfee` is not between 0-10|400|"listingfee"|Value sent in, e.g. "20" |"Listing fee can only be between 0 and 10"|
+|Invalid fee value|max_fee format is not valid|400|"max_fee"|Value sent in, e.g. "-100"|"Invalid fee value"|
+|Fee exceeds maximum|Actual fee is greater than supplied max_fee|400|"max_fee"|Value sent in, e.g. "1000000000"|"Fee exceeds supplied maximum"|
+|Insufficient funds to cover fee|Account does not have enough funds to cover fee|400|"max_fee"|Value sent in, e.g. "1000000000"|"Insufficient funds to cover fee"|
+|Invalid TPID|tpid format is not valid|400|"tpid"|Value sent in, e.g. "notvalidfioaddress"|"TPID must be empty or valid FIO address"|
 |Actor not signer|The signer and the `actor` do not match |403| | |Type: invalid_signature|
 
-#### Response
-TBD 
+#### Response body
+|Parameter|Format|Definition|
+|---|---|---|
+|status|String|OK if successful|
+|fee_collected|Int|Amount of SUFs collected as fee|
 
 ##### Example
-TBD
-
+```
+{
+  "status": "OK",
+  "fee_collected": 2000000000
+}
+```
 ### Remove Marketplace Config
-
+#### Contract: fio.escrow
+#### New fee: remove_marketplace_config, not bundle eligible
+#### New Endpoint: remove_marketplace_config
+#### Ram Increase: To be determined during implementation
 #### New action: *rmmrkplcfg*
-This removes a marketplace configuration. A marketplace configuration cannot be removed if there are any listings associated with this marketplace.
+This removes a marketplace configuration. 
+
+This can only be called by the BPs via msig. 
 
 #### Request
 |Parameter|Required|Format|Definition|
 |---|---|---|---|
 |actor|Yes|12 character string|Valid actor of signer |
 |marketplace|Yes|1-25 character string|Valid, registered marketplace|
+|max_fee|Yes|Positive Int|Maximum amount of SUFs the user is willing to pay for fee. Should be preceded by [/get_fee](https://developers.fioprotocol.io/api/api-spec/reference/get-fee/get-fee) for correct value.|
+|tpid|Yes|FIO Address|FIO Address of the entity which generates this transaction. TPID rewards will be paid to this address. Set to empty if not known.|
```

Comment:
Same as above, don't think we need fee/TPID if callable only by BPs.

Also, what happens if all configs are removed?

Reply:
I will remove this entire action. The SetMarketplaceConfig and UpdateMarketplaceConfig can take care of changing the marketplace config.

this was left over when it was designed to support multiple marketplaces but for this initial version we are not supporting multiple marketplaces.